deckhouse-controller/pkg/controller/deckhouse-release/check_release.go

```diff
@@ -208,20 +207,42 @@ releaseLoop:
 					notificationShiftTime = &metav1.Time{Time: *release.GetApplyAfter()}
 				}
 			}
-			if err := releaseChecker.StepByStepUpdate(release.GetVersion(), newSemver); err != nil {
-				releaseChecker.logger.Errorf("step by step update failed. err: %v", err)
-				labels := map[string]string{
-					"version": release.GetVersion().Original(),
+
+			actual := release.GetVersion()
+			for !actual.Equal(newSemver) {
+				if actual, err = releaseChecker.StepByStepUpdate(actual, newSemver); err != nil {
+					releaseChecker.logger.Errorf("step by step update failed. err: %v", err)
+					labels := map[string]string{
+						"version": release.GetVersion().Original(),
+					}
+
+					r.metricStorage.GroupedVault.GaugeSet(metricUpdatingFailedGroup, "d8_updating_is_failed", 1, labels)
+					return err
 				}
 
-				r.metricStorage.GroupedVault.GaugeSet(metricUpdatingFailedGroup, "d8_updating_is_failed", 1, labels)
-				return err
-			}
+				err = r.createRelease(ctx, releaseChecker, cooldownUntil, notificationShiftTime)
+				if err != nil {
+					return fmt.Errorf("crate release %s: %w", releaseChecker.releaseMetadata.Version, err)
+				}
 
-			break releaseLoop
+				if releaseChecker.releaseMetadata.Cooldown != nil {
+					cooldownUntil = releaseChecker.releaseMetadata.Cooldown
+				}
+			}
 		}
 	}
 
+	err = r.createRelease(ctx, releaseChecker, cooldownUntil, notificationShiftTime)
+	if err != nil {
+		return fmt.Errorf("crate release %s: %w", releaseChecker.releaseMetadata.Version, err)
+	}
```

Comment:
```suggestion
	// if there are no releases in the cluster, we apply the latest release
	if len(pointerReleases) == 0 {
		err = r.createRelease(ctx, releaseChecker, cooldownUntil, notificationShiftTime)
		if err != nil {
			return fmt.Errorf("create release %s: %w", releaseChecker.releaseMetadata.Version, err)
		}
	}
```

Reply:
it's a rather NIT: I noticed that the newSemVer release is created twice (first time in the pointerRelease loop)